### PR TITLE
Fix "Module x.js did not instantiate" when name passed to System.register()

### LINKED ISSUE #1884: Error: Module x.js did not instantiate when name passed to System.register()
Hi! I'm using `s.js` along with `extras/named-register.js` to load a SystemJS bundle generated by rollup.

The bundle looks like this:

```js
System.register('Example', ['./dependencies-4522e8b7.js', './react-editable-svg-label-201c3210.js'], function (exports, module) {
  'use strict';
  var React, PropTypes, ReactDOM, _extends, Portal;
  ...
```

I'm getting this stack trace:

```
s.js:171 Uncaught (in promise) Error: Module http://localhost:5000/example.js did not instantiate
    at s.js:171
(anonymous) @ s.js:171
load (async)
(anonymous) @ s.js:381
systemJSPrototype.instantiate @ s.js:373
systemJSPrototype.instantiate @ named-register.js:39
(anonymous) @ s.js:167
Promise.then (async)
getOrCreateLoad @ s.js:166
(anonymous) @ s.js:129
Promise.then (async)
systemJSPrototype.import @ s.js:128
(anonymous) @ (index):22
```

The problem seems to be that the plugin's `register()` doesn't invoke the base register function.

https://github.com/systemjs/systemjs/blob/fecfa650fd2042c8c4b758a50cc1470e8f5633d9/src/extras/named-register.js#L24

As a result, `lastRegister` is not set, and `loader.getRegister()` returns `undefined`.

https://github.com/systemjs/systemjs/blob/a0b95a50830ff97b58b29f19b885a058605fbe4c/src/system-core.js#L49-L61

I'll open a PR with the change that fixes this for me.

I'm new to Rollup and SystemJS, so if I'm misunderstanding how this is supposed to work, please do explain! 😝

### PR DESCRIPTION
**Pull request by paulmelnikow**
Fix #1884

### CODE REVIEW

**Comment by guybedford** [comment]
Thanks for the fix here, so the idea is that the named registration should just be an "empty module" and not the "last named registered module" in the bundle. As otherwise it's kind of arbitrary, although we could come up with a consistent convention there.

Could you please also add a test for this?

**Comment by guybedford** [comment]
Sure, I've responded on gitter.

File: src/extras/named-register.js [hunk 1]
```diff
@@ -20,8 +20,9 @@
   systemJSPrototype.register = function (name, deps, declare) {
     if (typeof name !== 'string')
       return register.apply(this, arguments);
-    
+
     this.registerRegistry[name] = [deps, declare];
+    return register(deps, declare);
```
[the remark this answers is not in the document]
Comment (paulmelnikow) [reply src/extras/named-register.js]: Makes sense… more or less, anyway!

[the remark this answers is not in the document]
Comment (paulmelnikow) [reply src/extras/named-register.js]: How do I get the actual module to execute?

[the remark this answers is not in the document]
Comment (paulmelnikow) [reply src/extras/named-register.js]: Thanks, that's really helpful. I'll try to do that and make sure this edit works and then update the PR.

**Comment by paulmelnikow** [comment]
Is there anything else I can do to help get this merged?

**Comment by paulmelnikow** [comment]
Hey @guybedford, could this be merged and released? I've pulling it in via git but am running into some toolchain issues on Zeit Now.

**Comment by guybedford** [comment]
Thanks again for the PR here, and apologies for the delay :) Will aim to release shortly.

**Comment by guybedford** [comment]
Published in 2.1.2.

**Comment by paulmelnikow** [comment]
Thank you!

**Pull request merged**